indico/modules/events/editing/client/js/editing/timeline/util.js

```diff
@@ -7,8 +7,44 @@
 
 import PropTypes from 'prop-types';
 
+import {FinalRevisionState, InitialRevisionState} from 'indico/modules/events/editing/models';
+import {Translate} from 'indico/react/i18n';
+
 import {filePropTypes} from './FileManager/util';
 
+// The top-down order defines a description for each state transition in a revision
+// [initialState, finalState] -> value | method -> value
+export const revisionStates = {
+  [InitialRevisionState.needs_submitter_confirmation]: {
+    [FinalRevisionState.accepted]: revision => {
+      return revision.editor !== null
+        ? Translate.string('Editor has accepted after making some changes')
+        : Translate.string('Submitter has accepted proposed changes');
+    },
+    [FinalRevisionState.needs_submitter_changes]: Translate.string(
+      'Submitter rejected proposed changes'
+    ),
+  },
+  any: {
+    [FinalRevisionState.replaced]: Translate.string('Revision has been replaced'),
+    [FinalRevisionState.accepted]: Translate.string('Revision has been accepted'),
+    [FinalRevisionState.rejected]: Translate.string('Revision has been rejected'),
+    [FinalRevisionState.needs_submitter_changes]: Translate.string(
+      'Submitter has been asked to make some changes'
+    ),
+    [FinalRevisionState.needs_submitter_confirmation]: revision =>
+      Translate.string('{editorName} (editor) has made some changes to the paper', {
+        editorName: revision.editor ? revision.editor.fullName : '',
```

Comment:
Would this work here?

```suggestion
        editorName: revision.editor?.fullName || '',
```

Reply:
Yes, sounds good 👍